ambassador/ambassador/ir/irauth.py

```diff
@@ -11,24 +11,44 @@
     from .ir import IR
 
 
+# Static allowed request headers normally used in the context of 
+# authorization and authentication. 
+AllowedRequestHeader = set([
+    'Authorization',
+    'Cookie',
+    'From',
+    'Host',
+    'Proxy-Authorization',
+    'User-Agent',
+    'X-Forwarded-For',
+    'X-Forwarded-Host',
+    'X-Forwarded-Proto'
+])
+
+# Static allowed authorization headers normally used in the context 
+# of authorization and authentication.
+AllowedAuthorizationHeaders = set([
+    'Location',
+    'Proxy-Authenticate',
+    'Set-Cookie',
+    'WWW-Authenticate'
+])
+
 class IRAuth (IRFilter):
     def __init__(self, ir: 'IR', aconf: Config,
                  rkey: str="ir.auth",
                  kind: str="IRAuth",
-                 name: str="extauth",
-                 type: Optional[str] = "decoder",
+                 name: str="envoy.ext_authz",
```

Comment:
We haven't ripped out the Envoy V1 config, so probably the `name` and `type` should stay as before.

Reply:
Not sure how `extauth` won't work in V2.

Reply:
That was, uh, a very good question to which I'd forgotten the answer. :smile:

So! the auth filter config is generated by `v1listener` or `v2listener`, depending on the config version. The `v1listener` code, as written, uses the `name` of the `IRAuth` object as the name for the Envoy filter. The `v2listener` code, though, hardcodes `envoy.extauth`.

Since the Envoy filter name is (obviously :smile:) very Envoy-specific, I feel like the right thing is to let the `listener` generates insert the correct thing for their version of Envoy, ignoring the `name` in the `IRAuth` object, rather than ever requiring the `IRAuth` object `name` to match an Envoy filter name.

You'll need to follow the code path by which the `IRAuth` is initialized to see if it's safe to rip out `type` for both the `ambassador/v0` and `ambassador/v1` API versions. I suspect that it has to stay in for the `ambassador/v0` API version to keep working, though of course the `listener` generators can ignore it, too, in favor of doing what's right for their Envoy version.

Reply:
Thanks for the explanation. I will work on this and I let you know if I have more questions.

Reply:
I tried to do these changes and it's just impossible to get this to work. It seems that IRauth is being used to support a bunch of legacy stuff which makes everything really confusing. I'm not entirely sure how Ambassador decides between V1 an V2 configs, or where IR is getting initialized.

Reply:
OK, let's jump on a call this morning and try to sort this out with screen sharing or something?

Reply:
Thanks @kflynn for kindly offering it. You said in later comment that this is out of date and I think it is: https://github.com/datawire/ambassador/blob/release/0.50.0/ambassador/README.md
 
Any other place in code that could help me to understand what is going on?